src/com/palmergames/bukkit/towny/command/TownyAdminCommand.java

```diff
@@ -1268,4 +1276,227 @@ public void parseToggleCommand(String[] split) throws TownyException {
+					player.sendMessage(field.getKey() + " = " + field.getValue());
+				}
+			} else {
+				TownyMessaging.sendErrorMsg(player, "This town doesn't have any associated metadata");
+			}
+
+			return;
+		}
+
+		if (split.length < 4) {
+			player.sendMessage(ChatTools.formatCommand("", "meta", "set", "The key of a registered data field"));
+			player.sendMessage(ChatTools.formatCommand("", "meta", "add", "Add a key of a registered data field"));
+			return;
+		}
+
+		if (split.length == 5) {
+			String operation = split[2];
+			String mdKey = split[3];
+			String val = split[4];
+
+			if (!townyUniverse.getRegisteredMetadataMap().containsKey(mdKey)){
+				TownyMessaging.sendErrorMsg(player, "The metadata for " + "\"" + mdKey + "\"" + " is not registered!");
+				return;
+			} else if (split[2].equalsIgnoreCase("set")) {
+				if (!player.hasPermission("towny.meta.modify")) {
+					TownyMessaging.sendErrorMsg("You don't have permissions for this.");
+					return;
+				}
+				CustomDataField md = townyUniverse.getRegisteredMetadataMap().get(mdKey);
+				if (town.hasMeta()) {
+					for (CustomDataField cdf: town.getMetadata()) {
+						if (cdf.equals(md)) {
+
+							// Check if the given value is valid for this field.
+							try {
+								cdf.isValidType(val);
+							} catch (InvalidMetadataTypeException e) {
+								TownyMessaging.sendErrorMsg(player, e.getMessage());
+								return;
+							}
+							
+							// Change state TODO: Add type casting..
+							cdf.setValue(val);
+
+							// Let user know that it was successful.
+							TownyMessaging.sendMessage(player, ChatColor.GREEN + "Key " + mdKey + " was successfully updated to " + cdf.getValue());
+
+							// Save changes.
+							townyUniverse.getDataSource().saveTown(town);
+
+							return;
+						}
+					}
+				}
+
+				TownyMessaging.sendErrorMsg(player, ChatColor.GREEN + "Key " + mdKey + " is not part of this town, if possible please add it");
+
+				return;
+
+			}
+		} else if (split[2].equalsIgnoreCase("add")) {
+			String mdKey = split[3];
+
+			if (!townyUniverse.getRegisteredMetadataMap().containsKey(mdKey)) {
+				TownyMessaging.sendErrorMsg(player, "The metadata for " + "\"" + mdKey + "\"" + " is not registered!");
+				return;
+			}
+
+			CustomDataField md = townyUniverse.getRegisteredMetadataMap().get(mdKey);
+
+			if (town.hasMeta()) {
+				for (CustomDataField cdf : town.getMetadata()) {
+					if (cdf.equals(md)) {
+						TownyMessaging.sendErrorMsg(player, ChatColor.GREEN + "Key " + mdKey + " already exists!");
+						return;
+					}
+				}
+			}
+
+			player.sendMessage(ChatColor.GREEN + "Custom data was successfully added to townblock!");
+
+			town.addMetaData(md);
+
+			// Save.
+			townyUniverse.getDataSource().saveTown(town);
+		}
+	}
+	
+	public static boolean handlePlotMetaCommand(Player player, String[] split) {
+		
+		String world = player.getWorld().getName();
+		TownBlock townBlock = null;
+		TownyUniverse townyUniverse = TownyUniverse.getInstance();
+		

```

Comment:
Add single permission node test here:

Change permission test over to our standard:
if (!townyUniverse.getPermissionSource().testPermission(player, PermissionNodes.TOWNY_COMMAND_TOWNYADMIN_PLOT_META.getNode()))
				throw new TownyException(TownySettings.getLangString("msg_err_command_disable"));